### PR TITLE
Add BulkDownload model.

### PR DESCRIPTION
**Pull request by MarkAZhang**
# Tests

* Verified in Rails console that BulkDownload object works as expected.

### CODE REVIEW

**Comment by MarkAZhang** [comment]
Requesting @gregdingle's review he worked on foreign keys.

File: db/migrate/20191009004932_create_bulk_downloads_table.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class CreateBulkDownloadsTable < ActiveRecord::Migration[5.1]
+  def change
+    create_table :bulk_downloads do |t|
+      t.text "params_json", comment: "JSON of the params for this bulk download"
```
Comment (gregdingle) [inline db/migrate/20191009004932_create_bulk_downloads_table.rb]: love the comments 😉

File: db/migrate/20191009004932_create_bulk_downloads_table.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      t.string "download_type", comment: "The type of bulk download"
```
Comment (gregdingle) [inline db/migrate/20191009004932_create_bulk_downloads_table.rb]: if we want to want to get really fancy, we could use rails enum types here, and for `status`. disclaimer: I have not used them before.

File: db/migrate/20191009004932_create_bulk_downloads_table.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      t.string "status", comment: "The current status of the download, e.g. waiting, running, error, success"
+      t.string "error_message", comment: "An error message to display to the user."
+    end
+
+    create_join_table :pipeline_runs, :bulk_downloads do |t|
+      t.index [:pipeline_run_id]
+      t.index [:bulk_download_id]
+    end
+
+    add_foreign_key "bulk_downloads_pipeline_runs", :bulk_downloads, name: "bulk_downloads_pipeline_runs_bulk_download_id_fk"
```
Comment (gregdingle) [inline db/migrate/20191009004932_create_bulk_downloads_table.rb]: 👍

File: db/migrate/20191009004932_create_bulk_downloads_table.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+class CreateBulkDownloadsTable < ActiveRecord::Migration[5.1]
+  def change
+    create_table :bulk_downloads do |t|
+      t.text "params_json", comment: "JSON of the params for this bulk download"
+      t.string "download_type", comment: "The type of bulk download"
+      t.string "status", comment: "The current status of the download, e.g. waiting, running, error, success"
+      t.string "error_message", comment: "An error message to display to the user."
```
Comment (gregdingle) [inline db/migrate/20191009004932_create_bulk_downloads_table.rb]: it seems like every column except this one should be not null. if so, please add `null: false` to the definitions.

File: db/migrate/20191009004932_create_bulk_downloads_table.rb (hunk 2 above)
Comment (MarkAZhang) [reply db/migrate/20191009004932_create_bulk_downloads_table.rb]: My main concern, apart from us having never used it before, is that enums get stored as numbers in the database, which makes using the Rails console much more annoying. (you have to remember what "0" and "1" mean) Will defer for now.

File: db/migrate/20191009004932_create_bulk_downloads_table.rb (hunk 1 above)
Comment (gregdingle) [inline db/migrate/20191009004932_create_bulk_downloads_table.rb]: so `params_json` may be null?

Comment (MarkAZhang) [reply db/migrate/20191009004932_create_bulk_downloads_table.rb]: Yeah. For some types of downloads there are no params. We could require everything to have an empty json `{}`, or just interpret `null` as `{}`.

**Comment by MarkAZhang** [comment]
Okay, I will look into the ActiveRecord way of defining enum as I flesh out the model more. Thanks!

**Pull request merged**